src/xSTIR/cSTIR/include/sirf/STIR/stir_x.h

```diff
@@ -911,13 +915,33 @@ The actual algorithm is described in
 			sptr_ad_ = sptr;
 			set_proj_data_sptr(sptr->data());
 		}
-		void set_acquisition_model(stir::shared_ptr<AcqMod3DF> sptr)
+		void set_acquisition_model(stir::shared_ptr<AcqMod3DF> sptr_am)
 		{
-			sptr_am_ = sptr;
-			AcqMod3DF& am = *sptr;
+			sptr_am_ = sptr_am;
+			AcqMod3DF& am = *sptr_am;
+			stir::shared_ptr<PETAcquisitionSensitivityModel> sptr_asm = am.asm_sptr();
```

Comment:
I'd prefer to write these things as
```suggestion
			auto sptr_asm = am.asm_sptr();
```
as this is shorter and makes us less sensitive to `shared_ptr` choices etc

Reply:
ok